src/lib/src/datepicker/timepicker.component.ts

```diff
@@ -16,26 +16,31 @@ export class TimepickerComponent implements OnChanges {
   @Input() date: Date;
   @Output() timeChange = new EventEmitter<Date>();
   @Input() changed: boolean;
+  @Input() UTC = false;
 
-  localTimezone: string;
+  timezone: string;
   options: string[];
 
   ngOnChanges() {
     let dayGen = new Date(1970, 0, 1, 0, 0, 0, 0);
-    let match = this.date ? this.date.toString().match(/(\([A-Za-z\s].*\))/) : dayGen.toString().match(/(\([A-Za-z\s].*\))/);
-    if (match && match.length > 0) {
-      this.localTimezone = match[1];
+    if (this.UTC) {
+      this.timezone = '(GMT)';
+    } else {
+      let match = this.date ? this.date.toString().match(/(\([A-Za-z\s].*\))/) : dayGen.toString().match(/(\([A-Za-z\s].*\))/);
+      if (match && match.length > 0) {
+        this.timezone = match[1];
+      }
     }
     let day = dayGen.getDate();
     this.options = [];
     while (dayGen.getDate() === day) {
-      this.options.push(this.dateToHumanTime(dayGen));
+      this.options.push(this.dateToHumanTime(dayGen, false));
       dayGen.setMinutes(dayGen.getMinutes() + 30);
     }
   }
 
   roundMinutes(value: number): string {
-    return value > 0 && value <= 30 ? '30' : '00';
+    return value >= 0 && value < 30 ? '00' : '30';
```

Comment:
This bug never affected Publish that I'm aware of because times are X:00 or X:30. I think my original intention was round up, but 11:59pm was rounding down to 11:00pm. I didn't fix it to properly round up because I don't want to go into the next day, so it rounds down.